executor/insert_common.go

```diff
@@ -188,9 +190,17 @@ func (e *InsertValues) insertRows(ctx context.Context, exec func(ctx context.Con
 	rows := make([][]types.Datum, 0, len(e.Lists))
 	for i, list := range e.Lists {
 		e.rowCount++
-		row, err := e.evalRow(ctx, list, i)
-		if err != nil {
-			return err
+		var row []types.Datum
+		if e.notAllConstant {
```

Comment:
Try to avoid branch prediction failure.
```go
    // put below out of loop.
	evalRowFunc := e.fastEvalRow
	if e.notAllConstant {
		evalRowFunc = e.evalRow	
	}
```